src/stub/core_impl/base.rs

```diff
@@ -302,17 +306,26 @@ impl<T: Target, C: Connection> GdbStubImpl<T, C> {
                 use crate::protocol::commands::_h_upcase::Op;
                 match cmd.kind {
                     Op::Other => match cmd.thread.tid {
-                        IdKind::Any => self.current_mem_tid = self.get_sane_any_tid(target)?,
+                        IdKind::Any => match self.get_sane_any_tid(target)? {
+                            Some(tid) => self.current_mem_tid = tid,
+                            None => {
+                                res.write_str("E01")?;
+                                return Ok(HandlerStatus::Handled);
+                            }
```

Comment:
I actually have a bit of plumbing in place to streamline returning `Exx`-style errors:

```suggestion
                            None => return Err(Error::NonFatalError(1))
```

It's both more concise than invoking write_str + returning early, and also results in some slightly better codegen (wrt. binary size).

Can you switch both blocks over?

Reply:
Sure, I've made this change and pushed the result.